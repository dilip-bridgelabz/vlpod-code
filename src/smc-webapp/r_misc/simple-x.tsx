/*
 *  This file is part of CoCalc: Copyright © 2020 Sagemath, Inc.
 *  License: AGPLv3 s.t. "Commons Clause" – see LICENSE.md for details
 */

import * as React from "react";
import { Icon } from "./icon";

interface Props {
  onClick: () => void;
}

export function SimpleX({ onClick }: Props) {
  return (
    <a
      href=""
      onClick={(e) => {
        e.preventDefault();
        onClick();
      }}
    >
      <Icon name="times" />
    </a>
  );
}
