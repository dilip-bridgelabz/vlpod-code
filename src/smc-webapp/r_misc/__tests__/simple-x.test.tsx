/*
 *  This file is part of CoCalc: Copyright © 2020 Sagemath, Inc.
 *  License: AGPLv3 s.t. "Commons Clause" – see LICENSE.md for details
 */

import * as React from "react";
import { shallow } from "enzyme";
import { SimpleX } from "../simple-x";

test("smoke test", () => {
  const rendered = shallow(<SimpleX onClick={() => undefined} />);
  expect(rendered).toMatchSnapshot();
});

test("test interaction", () => {
  const mock_on_click = jest.fn();
  const rendered = shallow(<SimpleX onClick={mock_on_click} />);

  const evt = new MouseEvent("click");

  rendered.simulate("click", evt);
  expect(mock_on_click.mock.calls.length).toBe(1);
});
