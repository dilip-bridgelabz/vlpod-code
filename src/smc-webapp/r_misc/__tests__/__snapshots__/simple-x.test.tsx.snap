// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke test 1`] = `
<a
  href=""
  onClick={[Function]}
>
  <Icon
    name="times"
  />
</a>
`;
